### PR TITLE
Balinus/patch 1 savedoc

### PR DESCRIPTION
**Pull request by Balinus**
Documentation about compression #215

### CODE REVIEW

File: docs/src/UserGuide/write.md [hunk 1]
```diff
@@ -28,6 +28,17 @@ savedataset(ds, path="ds.zarr", driver=:zarr)
 nothing # hide
 ````
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:30]:
```suggestion
### Compression
```
we should add this as a new sub-section.

File: docs/src/UserGuide/write.md [hunk 2]
```diff
@@ -28,6 +28,17 @@ savedataset(ds, path="ds.zarr", driver=:zarr)
...
 
+Save a dataset to Zarr format with compression:
+
+````@example write
+using Blosc
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:34]: you need to add `Blosc` to the Project toml, explicitly. Or, loaded it from whatever other package is also available.

File: docs/src/UserGuide/write.md [hunk 3]
```diff
@@ -28,6 +28,17 @@ savedataset(ds, path="ds.zarr", driver=:zarr)
...
+n = 7 # compression level, number between 0 (no compression) and 9 (max compression)
+compression = Zarr.BloscCompressor(clevel=n)
+
+savedataset(ds, path="ds.zarr", driver=:zarr, compressor=compression)
+nothing # hide
+````
+
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:41]: we should have more than 1 compression level, and compare them both at the end, namely, what their disk sizes.

File: docs/src/UserGuide/write.md [hunk 4]
```diff
@@ -28,6 +28,17 @@ savedataset(ds, path="ds.zarr", driver=:zarr)
 nothing # hide
 ````
 
+Save a dataset to Zarr format with compression:
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:31]:
```suggestion
#### Saving to zarr
Save a dataset to Zarr format with compression:
```

File: docs/src/UserGuide/write.md [hunk 5]
```diff
@@ -28,6 +28,17 @@ savedataset(ds, path="ds.zarr", driver=:zarr)
...
+````@example write
+using Blosc
+n = 7 # compression level, number between 0 (no compression) and 9 (max compression)
+compression = Zarr.BloscCompressor(clevel=n)
+
+savedataset(ds, path="ds.zarr", driver=:zarr, compressor=compression)
+nothing # hide
+````
+
 ## Write NetCDF
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:42]:
```suggestion
#### Write NetCDF
```

File: docs/src/UserGuide/write.md [hunk 6]
```diff
@@ -45,6 +56,14 @@ savedataset(ds, path="ds.nc", driver=:netcdf)
 nothing # hide
 ````
 
+Save a dataset to NetCDF format with compression:
+
+````@example write
+n = 7 # compression level, number between 0 (no compression) and 9 (max compression)
+savedataset(ds, path="ds.zarr", driver=:zarr, compress=n)
```
Comment (lazarusA) [inline docs/src/UserGuide/write.md:63]:
```suggestion
savedataset(ds, path="ds.nc", driver=:netcdf, compress=n)
```

**Review by lazarusA — CHANGES_REQUESTED** [review]
thanks a lot! some minor comments.

File: docs/src/UserGuide/write.md (hunk 3 above)
Comment (Balinus) [reply docs/src/UserGuide/write.md:41]: You mean, compare on my computer and report sizes in the documentation?

File: docs/src/UserGuide/write.md (hunk 2 above)
Comment (Balinus) [reply docs/src/UserGuide/write.md:34]: Adding Blosc to YAXArrays's Project.toml? or perhaps you mean YAXArrays/docs/Project.toml ?

Comment (lazarusA) [reply docs/src/UserGuide/write.md:34]: to docs 😄 .

File: docs/src/UserGuide/write.md (hunk 3 above)
Comment (lazarusA) [reply docs/src/UserGuide/write.md:41]: here, directly. Let's say a `ds_level_1.zarr` and `ds_level4.zarr`.  And then query their sizes and print them.

Comment (lazarusA) [reply docs/src/UserGuide/write.md:41]: I had never use this argument, so..., also I don't know what are the defaults, and hence how different things actually are.

Comment (Balinus) [reply docs/src/UserGuide/write.md:41]: oh, I see I didn't even thought that the code were actually run! We'll look into it

File: docs/src/UserGuide/write.md (hunk 2 above)
Comment (Balinus) [reply docs/src/UserGuide/write.md:34]: I think we can access the compressor within Zarr.jl directly. I will adapt the code

File: docs/src/UserGuide/write.md (hunk 3 above)
Comment (Balinus) [reply docs/src/UserGuide/write.md:41]: Humm, strangely, the compression does not seems to work as expected. There is no effect on Zarr folders and there is some on netcdf files...

Comment (Balinus) [reply docs/src/UserGuide/write.md:41]: 76M	test1.nc
103M test_default.nc
75M	test9.nc

84M	test1.zarr
83M	test_default.zarr
82M	test9.zarr

**Pull request merged**